app/src/main/java/com/cp/tasky/auth/login/presentation/LoginViewModel.kt

```diff
@@ -46,18 +45,18 @@ class LoginViewModel @Inject constructor(
     }
 
     private fun loginUser(email: String, password: String) {
-        _networkState.value = Result.Loading
-
         viewModelScope.launch {
+            _networkResponse.send(Result.Loading)
+
             authRepository.loginUser(
                 UserCredential(
                     email = email,
                     password = password
                 )
-            ).onSuccess { responseData ->
-                _networkState.value = Result.Success(responseData)
+            ).onSuccess {
+                _networkResponse.send(Result.Success(Unit))
```

Comment:
This one is fine, since you just care about navigating here which you wouldn't want to redo after the user rotates the device or so.